modules/deployments-ab-testing-lb.adoc

```diff
@@ -0,0 +1,258 @@
+// Module included in the following assemblies:
+//
+// * applications_and_projects/deployments/route-based-deployment-strategies.adoc
+
+[id='deployments-ab-testing-lb-{context}']
+= Load balancing for A/B testing
+
+The user sets up a route with multiple services. Each service handles a version
+of the application.
+
+Each service is assigned a `weight` and the portion of requests to each service
+is the `service_weight` divided by the `sum_of_weights`. The `weight` for each
+service is distributed to the service's endpoints so that the sum of the
+endpoint `weights` is the service `weight`.
+
+The route can have up to four services. The `weight` for the service can be
+between `0` and `256`. When the `weight` is `0`, the service does not participate in load-balancing
+but continues to serve existing persistent connections. When the service `weight`
+is not `0`, each endpoint has a minimum `weight` of `1`. Because of this, a
+service with a lot of endpoints can end up with higher `weight` than desired.
+In this case, reduce the number of Pods to get the desired load balance
+`weight`.
+
+////
+See the
+xref:../../architecture/networking/routes.adoc#alternateBackends[Alternate
+Backends and Weights] section for more information.
+
+The web console allows users to set the weighting and show balance between them:
+
+image::weighting.png[Visualization of Alternate Back Ends in the Web Console]
+////
+
+.Procedure
+
+To set up the A/B environment:
+
+. Create the two applications and give them different names. Each creates a
+DeploymentConfig. The applications are versions of the same program; one
+is usually the current production version and the other the proposed new
+version:
++
+----
+$ oc new-app openshift/deployment-example --name=ab-example-a
+$ oc new-app openshift/deployment-example --name=ab-example-b
+----
++
+Both applications are deployed and services are created.
+
+. Make the application available externally via a route. You can expose either
+service at this point, it may be convenient to expose the current production
+version first and later modify the route to add the new version.
```

Comment:
At this point, you can expose either service. I can be convenient to expose the current production version first, then later...